diff -cr /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnat/a-elchha.adb ./gnat/a-elchha.adb
*** /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnat/a-elchha.adb	2025-04-03 17:37:10.233859352 +0200
--- ./gnat/a-elchha.adb	2023-01-09 14:28:17.000000000 +0100
***************
*** 6,12 ****
  --                                                                          --
  --                                 B o d y                                  --
  --                                                                          --
! --          Copyright (C) 2003-2021, Free Software Foundation, Inc.         --
  --                                                                          --
  -- GNAT is free software;  you can  redistribute it  and/or modify it under --
  -- terms of the  GNU General Public License as published  by the Free Soft- --
--- 6,12 ----
  --                                                                          --
  --                                 B o d y                                  --
  --                                                                          --
! --          Copyright (C) 2003-2020, Free Software Foundation, Inc.         --
  --                                                                          --
  -- GNAT is free software;  you can  redistribute it  and/or modify it under --
  -- terms of the  GNU General Public License as published  by the Free Soft- --
***************
*** 29,69 ****
  --                                                                          --
  ------------------------------------------------------------------------------
  
! with System.IO;                use System.IO;
! with GNAT.Debug_Utilities;     use GNAT.Debug_Utilities;
! with System;
! with System.OS_Interface;
  
  --  Default last chance handler for use with full run-time library on bare
  --  board targets.
  
  --  It dumps the exception identity and the non-symbolic traceback from the
  --  point where the exception was triggered.
- 
  procedure Ada.Exceptions.Last_Chance_Handler (Except : Exception_Occurrence) is
! begin
!    Put_Line ("In last chance handler");
!    Put_Line ("Unhandled Ada Exception: " & Exception_Name (Except));
!    Put_Line ("Message: " & Exception_Message (Except));
!    Put_Line ("Call stack traceback locations:");
  
!    --  Dump backtrace PC values
  
!    for J in 1 .. Except.Num_Tracebacks loop
!       Put (Image_C (Except.Tracebacks (J)));
!       Put (" ");
!    end loop;
  
!    New_Line;
  
!    --  Suspend forever
  
!    System.OS_Interface.Sleep;
  
!    --  The following junk raise of Program_Error is required because
!    --  this is a No_Return function, and unfortunately Suspend can
!    --  return (although this particular call won't).
  
!    raise Program_Error;
  
  end Ada.Exceptions.Last_Chance_Handler;
--- 29,138 ----
  --                                                                          --
  ------------------------------------------------------------------------------
  
! --  with System.IO;                use System.IO;
! --  with GNAT.Debug_Utilities;     use GNAT.Debug_Utilities;
! --  with System;
! --  with System.OS_Interface;
! 
! with Interfaces;
! with Interfaces.C.Strings;
  
  --  Default last chance handler for use with full run-time library on bare
  --  board targets.
  
  --  It dumps the exception identity and the non-symbolic traceback from the
  --  point where the exception was triggered.
  procedure Ada.Exceptions.Last_Chance_Handler (Except : Exception_Occurrence) is
!    type Pobj is access Integer;
! 
!    type ExceptionMessageBoxCallbackType is access
!      procedure (Obj : Pobj; Event : Interfaces.Unsigned_8);
!    pragma Convention (C, ExceptionMessageBoxCallbackType);
! 
!    ExceptionMessageBoxFlag : aliased Boolean := False;
!    pragma Volatile (ExceptionMessageBoxFlag);
! 
!    procedure ExceptionMessageBoxCallback (Obj : Pobj;
!                                           Event : Interfaces.Unsigned_8);
!    pragma Convention (C, ExceptionMessageBoxCallback);
  
!    procedure Reboot;
!    pragma Import (C, Reboot, "Reboot_System");
  
!    --  CreateExceptionMessageBox
!    --  Create a messagebox for information
!    procedure CreateExceptionMessageBox (Msg : String; Button_Txt : String);
  
!    procedure CreateExceptionMessageBox (Msg : String; Button_Txt : String) is
  
!       function Wrp_UI_MESSAGEBOX_Create
!         (Msg : Interfaces.C.Strings.chars_ptr;
!          Button_Txt : Interfaces.C.Strings.chars_ptr_array;
!          Callback : ExceptionMessageBoxCallbackType) return Pobj;
!       pragma Import (C, Wrp_UI_MESSAGEBOX_Create, "UI_MESSAGEBOX_Create");
  
!       Mbx : Pobj;
!       pragma Volatile (Mbx);
  
!       Msg_C : constant Interfaces.C.Strings.chars_ptr :=
!         Interfaces.C.Strings.New_String (Msg);
!       Button_Txt_Array : constant Interfaces.C.Strings.chars_ptr_array (0 .. 1)
!         := (Interfaces.C.Strings.New_String (Button_Txt),
!             Interfaces.C.Strings.New_String (""));
!    begin
!       ExceptionMessageBoxFlag := False;
  
!       Mbx := Wrp_UI_MESSAGEBOX_Create (Msg_C, Button_Txt_Array,
!                                        ExceptionMessageBoxCallback'Access);
  
+       --  while ExceptionMessageBoxFlag = False loop
+       --     null;
+       --  end loop;
+ 
+       --  Wrp_Lv_Msgbox_Close(Mbx, 50);
+    end CreateExceptionMessageBox;
+ 
+    --  messagebox callback
+    procedure ExceptionMessageBoxCallback (Obj : Pobj;
+                                           Event : Interfaces.Unsigned_8) is
+ 
+       pragma Unreferenced (Obj, Event);
+ 
+    begin
+       Reboot;
+    end ExceptionMessageBoxCallback;
+ 
+ begin
+    CreateExceptionMessageBox ("Unhandled Ada Exception: " & ASCII.LF &
+                                 Exception_Name (Except) &
+                                 ASCII.LF & ASCII.LF &
+                                 Exception_Message (Except),
+                               "Reboot");
+ 
+    --  Put_Line ("In last chance handler");
+    --  Put_Line ("Unhandled Ada Exception: " & Exception_Name (Except));
+    --  Put_Line ("Message: " & Exception_Message (Except));
+    --  Put_Line ("Call stack traceback locations:");
+    --
+    --  --  Dump backtrace PC values
+    --
+    --  for J in 1 .. Except.Num_Tracebacks loop
+    --     Put (Image_C (Except.Tracebacks (J)));
+    --     Put (" ");
+    --  end loop;
+    --
+    --  New_Line;
+    --
+    --  --  Suspend forever
+    --
+    --  System.OS_Interface.Sleep;
+    --
+    --  --  The following junk raise of Program_Error is required because
+    --  --  this is a No_Return function, and unfortunately Suspend can
+    --  --  return (although this particular call won't).
+    --
+    --  raise Program_Error;
+    pragma Warnings (Off, "*rewritten as loop");
+    <<Spin>> goto Spin;   -- yes, a goto!
+    pragma Warnings (On, "*rewritten as loop");
  end Ada.Exceptions.Last_Chance_Handler;
diff -cr /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnat/setup_pll.adb ./gnat/setup_pll.adb
*** /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnat/setup_pll.adb	2025-04-03 17:37:15.864874677 +0200
--- ./gnat/setup_pll.adb	2023-02-06 10:54:45.000000000 +0100
***************
*** 44,49 ****
--- 44,50 ----
  
  procedure Setup_Pll is
     procedure Initialize_Clocks;
+    pragma Unreferenced (Initialize_Clocks);
     procedure Reset_Clocks;
  
     ------------------------------
***************
*** 279,284 ****
--- 280,286 ----
           loop
              exit when System.BB.MCU_Parameters.Is_PWR_Stabilized;
           end loop;
+ 
        end if;
     end Initialize_Clocks;
  
***************
*** 311,315 ****
  
  begin
     Reset_Clocks;
!    Initialize_Clocks;
  end Setup_Pll;
--- 313,317 ----
  
  begin
     Reset_Clocks;
!    --  Initialize_Clocks;
  end Setup_Pll;
Seulement dans /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnat: start-common.S
Seulement dans /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnat: start-ram.S
Seulement dans /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnat: start-rom.S
Seulement dans ./gnat: startup_stm32f746xx.S
